tests/integration/replication-4.tcl

```diff
@@ -12,12 +12,23 @@ start_server {tags {"repl network external:skip"}} {
 
         test {First server should have role slave after SLAVEOF} {
             $slave slaveof $master_host $master_port
-            after 1000
-            s 0 role
-        } {slave}
+            wait_for_condition 50 100 {
+                [s 0 role] eq {slave}
+            } else {
+                fail "Replication not started."
+            }
+        }
 
         test {Test replication with parallel clients writing in different DBs} {
-            after 5000
```

Comment:
maybe we should let them do some work (not exit after 1 key).
these are "complex" loaders, using random commands, and we're later validating the database digest..
i think the right fix would be to keep both the wait_for you added, and the 5 second sleep (maybe add a comment it "gives the random workloads a chance to add some complex commands".

Reply:
yes, sound good, i am changing it